src/traces/box/calc.js

```diff
@@ -178,7 +178,10 @@ function getPos(trace, posLetter, posAxis, val, num) {
         pos0 = num;
     }
 
-    var pos0c = posAxis.d2c(pos0, 0, trace[posLetter + 'calendar']);
+    var pos0c = posAxis.type === 'multicategory' ?
+        posAxis.r2c(pos0) :
```

Comment:
Does it work to specify `x0: ['2016', 'day 1']` or does `x0` need to be a number? I'm not really sure the latter is something we want to encourage, and the former seems to me fairly important to support.

I see your mock with the violin halfway between the two inner categories, which of course isn't possible using the categories themselves, I feel like if someone really wants that we should add something like `boxpos`, ie an analog of `pointpos` to allow you to shift where the box is drawn, but still have it explicitly anchored to a specific category rather than to our serialized category numbers.

Reply:
> Does it work to specify `x0: ['2016', 'day 1']`

This works now in https://github.com/plotly/plotly.js/pull/3300/commits/5c8598121f6a04b6193e5aab65de6916701eb6af

Should we disallow number `x0` that falls in-between categories?

Reply:
> Should we disallow number `x0` that falls in-between categories?

Personally I'd be in favor of not allowing `x0` to be a serial number at all. It can be a number on a regular category axis, but that number is treated as a category. So allowing a serial number would be unique to `multicategory` and I suspect more likely confusing than useful.

Reply:
Are you ok with -> https://github.com/plotly/plotly.js/commit/27d74d8060a2a026ee08c531d24672d2df4af8c1

Reply:
done in https://github.com/plotly/plotly.js/pull/3300/commits/4b82426021d1e0957f45800a65a36328a5d6ba4a